### DISCUSSION TITLE
nuxt3中使用useFetch对同一个url发送多次不同请求体的POST请求时，是不是需要指定不同的key？

### DISCUSSION THREAD

**Discussion by GentleSevenV** [nuxt/nuxt#30162] [Questions] [answered]
nuxt3中使用useFetch对同一个url发送多次不同请求体的POST请求时，是不是需要指定不同的key？否则返回的数据都是一样的？

**Comment by xjccc** [comment]
多次调用会自动生成不同的key。看你怎么使用的？

**Reply by GentleSevenV** [reply]
`<script lang="ts" setup>
const res = await PostApi("/open/dict/info/data", {
  body: { types: ["headnav"] },
});
console.log(res);

const res2 = await PostApi("/open/dict/info/data", {
  body: { types: ["news"] },
});
console.log(res2);
</script>`

我是这样使用的，这里的PostApi是我封装的useFetch post请求，这里的2个请求url是相同的，唯一不同的是body中携带的types参数，但是当我打印这2个请求的结果时，res和res2都是获得了news的数据而没有headnav的数据，但是我尝试给每个请求手动指定不同key的时候，res和res2就可以分别获得了headnav和news的数据，所以我才说使用useFetch对同一个url发送多次不同请求体的POST请求时，是不是需要指定不同的key？否则返回的数据都是一样的？

**Reply by xjccc** [reply]
如果不是显式调用useFetch，那自动key生成的就是一样的，如果使用`watch`去触发呢？或者像你说的，最简单传入不同的`key`

**Reply by GentleSevenV** [reply]
刚才我按照你的方法尝试显示调用直接使用useFetch并且没有给这2个请求指定不同的key，就可以正常获得2个不同请求的结果。这样也验证了你说的显示调用的话，生成的key就是不一样的。使用watch这个方法，我之前尝试过，也是通过隐式调用，好像不起作用。但是简单的传入key是可以的。所以总结下来的话，useFetch隐式调用（对同一个url发送多次不同请求体的POST请求时）生成的key是一样的。但是不明白为什么会是这样？^_^

**Reply by xjccc** [reply, marked as answer]
`autoKey`是编译时注入的，你多次调用`useFetch`是通过调用行数文件名生成的，所以没问题。你封装后，他只有一个位置调用。

**Reply by GentleSevenV** [reply]
这下完全明白了，感谢大佬！！关注你了！
